### PR TITLE
[WB-3289] Kaggle internet check

### PR DESCRIPTION
**Pull request by jsbroks**
https://wandb.atlassian.net/browse/WB-3289
Throws an error if the user has not enabled internet in kaggle

### CODE REVIEW

**Comment by tssweeney** [comment]
Hey @jsbroks i have been trying to test this for a while now and can't figure out how to get the kaggle kernel to install this particular branch to test. Did you manage to get that working?

**Comment by jsbroks** [comment]
@tssweeney Yup here are the steps (I had to ask Jeff to help me with):

1. Turn the internet on under settings (its off by default).
2. ```
    !pip uninstall -y typing
    !pip install --upgrade git+git://github.com/wandb/client.git@justin/kaggle-warning
    ```
3. Turn internet off
4. ````
    import wandb
    wandb.init(project="testing")
    ```
You should see the following up:
image

**Review by tssweeney — APPROVED** [review]
Nice - validated the fix. LGTM

File: wandb/sdk/lib/server.py [hunk 1]
```diff
@@ -3,15 +3,25 @@
...
 
 class ServerError(Exception):
     pass
 
 
+def _is_kaggle() -> bool:
+    return (
+        os.getenv("KAGGLE_KERNEL_RUN_TYPE") is not None
+        or "kaggle_environments" in sys.modules  # noqa: W503
+    )
```
Comment (vanpelt) [inline wandb/sdk/lib/server.py:22]: We already have this code in `wandb_settings.py`.  Likely makes sense to put this method in `wandb.util` and share it in both places.

**Review by vanpelt — APPROVED** [review]

**Pull request merged**